### PR TITLE
NotificationData fix

### PR DESCRIPTION
**Pull request by danielh1204**
pr for https://github.com/free5gc/free5gc/issues/78

### CODE REVIEW

**Comment by danielh1204** [comment]
pr for https://github.com/free5gc/free5gc/issues/78

File: internal/sbi/producer/nf_management.go [hunk 1]
```diff
@@ -517,8 +517,48 @@ func NFRegisterProcedure(
 	}
 }
 
+func configureNotifNfProfile(NotifProfile *models.NfProfileNotificationData, nfProfile models.NfProfile) {
```
Comment (brianchennn) [inline internal/sbi/producer/nf_management.go:520]: func configureNotifNfProfile(NotifProfile *models.NfProfileNotificationData, nfProfile *models.NfProfile) {

Comment (tim-ywliu) [reply internal/sbi/producer/nf_management.go:520]: param `NotifProfile` rename to `notifProfile`

**Review by brianchennn — APPROVED** [review]

File: internal/sbi/producer/nf_management.go [hunk 2]
```diff
@@ -496,7 +496,7 @@ func NFRegisterProcedure(
 		nfInstanceUri := locationHeaderValue
 
 		for _, uri := range uriList {
-			problemDetails := SendNFStatusNotify(Notification_event, nfInstanceUri, uri)
+			problemDetails := SendNFStatusNotify(Notification_event, nfInstanceUri, uri, nil)
```
Comment (tim-ywliu) [inline internal/sbi/producer/nf_management.go:499]: why fill nil nfProfile here?

Comment (danielh1204) [reply internal/sbi/producer/nf_management.go:499]: in func SendNFStatusNotify, if the nfprofile is nil, it means i dont have to copy the nfprofile (the paramter) into the notificationData.NfProfile.
according to 3GPP spec
image

Comment (tim-ywliu) [reply internal/sbi/producer/nf_management.go:499]: but your nfProfile is nil and profileChanges is also not present.

Comment (danielh1204) [reply internal/sbi/producer/nf_management.go:499]: in the case of NFRegisterProcedure, the event attribute's value is REGISTERED when 'existed' returns as false at line 460

Comment (tim-ywliu) [reply internal/sbi/producer/nf_management.go:499]: 螢幕擷取畫面 2023-09-15 223209

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**